### PR TITLE
DIS-1763 - Error on unused exports

### PR DESCRIPTION
**Pull request by dbirks**
Use the [`import/no-unused-modules`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-unused-modules.md) rule to warn on unused exports. 

Making it apply to just `src/`, and excluding a bunch of spec and test files. I also am ignoring `src/pages/**` (I think excluding it is desired for Next apps, maybe?), so please tell me if you think something shouldn't be on that `excludedFiles` list, or if I could ignore more. :slightly_smiling_face: 

Tried consuming this in aurora-client (24 errors), horizon-pro-api (13 errors),  and polaris-ui-reference (5 errors), so it varies.
```
npm i --save-dev @polarislabs/eslint-config-polaris@2.3.0-beta.8
```

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -50,6 +50,29 @@ module.exports = {
...
+        "**/*.test.tsx",
+        "index.ts",
+        "index.tsx",
+        "main.ts",
+        "src/pages/**",
+      ],
+      files: ["src/**"],
+      rules: {
+        "import/no-unused-modules": [
+          "warn",
```
Comment (camoore1979) [inline common.js:68]: I _think_ this should be `"error"`. That's how we are enforcing most, if not all, of our ESLint rules. If necessary, an ignore can be added.

**Review by camoore1979 — CHANGES_REQUESTED** [review]

**Comment by camoore1979** [comment]
@dbirks Is there a specific beta version one can point to for evaluating this PR?

File: common.js (hunk 1 above)
Comment (dbirks) [reply common.js:68]: Ah sounds good, changing to `error`...

**Comment by dbirks** [comment]
Ok pushed up a new version, with the comments, changelog, and code all changed to `error`.

Tried in horizon-pro-api, and got the same results, but as errors as expected. Used this package:
```
npm i --save-dev @polarislabs/eslint-config-polaris@2.3.0-beta.8
```

**Review by camoore1979 — APPROVED** [review]

File: common.js [hunk 2]
```diff
@@ -50,6 +50,29 @@ module.exports = {
...
+      excludedFiles: [
+        "**/*.spec.ts",
+        "**/*.spec.tsx",
+        "**/*.test.ts",
+        "**/*.test.tsx",
+        "index.ts",
+        "index.tsx",
+        "main.ts",
+        "src/pages/**",
+      ],
```
Comment (babennettdev) [inline common.js:64]:
```suggestion
      excludedFiles: [
        "**/*.spec.ts",
        "**/*.spec.tsx",
        "**/*.stories.tsx",
        "**/*.test.ts",
        "**/*.test.tsx",
        "index.ts",
        "index.tsx",
        "main.ts",
        "src/pages/**",
      ],
```

**Review by babennettdev — CHANGES_REQUESTED** [review]
David, I have consumed this package locally in many of our repos, `HPAPI`, `aurora-client` and `service`, and `polaris-nest-core` and `polaris-components`. All reported varying results when running `npm run lint` depending on the number of unused exports. I think this will be a great opportunity to clean up our repos as we begin consuming this package.

My only recommendation is that we also exclude `**/*.stories.tsx` as well for `polaris-components` as these need to use default exports to run Storybook, but don't get consumed directly anywhere in the repository. All other eslint errors I saw seemed valid and will need to be rectified as we consume this version of the package in our repos.

File: common.js (hunk 2 above)
Comment (dbirks) [reply common.js]: Sounds good to me :+1: 
I'll get the changelog updated too to fix the merge conflict...

**Comment by dbirks** [comment]
I'm sorry, this repo is currently in an in-between state with some Github Actions for the release process. I'll finish up my changes to that today, and will re-add you both when it's ready for a final review.

**Comment by babennettdev** [comment]
Sounds good @dbirks let me know when this is ready for another look!

**Comment by dbirks** [comment]
Ok, this should be finally ready for a second look. I've got a new beta out as well, ignoring `**/*.stories.tsx`:
```
npm i --save-dev @polarislabs/eslint-config-polaris@2.3.0-beta.9
```

**Review by babennettdev — APPROVED** [review]
LGTM, David! Thank you for making the requested change!

Make sure you remove the `-beta.9` from the version before merge! 🚀

**Review by camoore1979 — APPROVED** [review]

**Comment by dbirks** [comment]
Thanks guys :+1: 
I just confirmed locally that running `npm version minor` bumps it up to 2.3.0 correctly, so I'll give it a merge...

**Pull request merged**